cores/arduino/HardwareTimer.cpp

```diff
@@ -474,6 +474,10 @@ void HardwareTimer::setOverflow(uint32_t overflow, TimerFormat_t format)
       break;
   }
 
+  // In case ARR computation give 0xFFFFFFFF (equal (0 - 1) ) set ARR register to 0 (the smallest achievable period)
+  if (ARR_RegisterValue == 0xFFFFFFFF) {
+    ARR_RegisterValue = 0;
+  }
```

Comment:
An alternative that might be easier to read here, would be to move the -1 down a bit. So let the switch set a `PeriodTicks` value without the -1, e.g.:

    PeriodTicks = (period_cyc / Prescalerfactor);

And then something like:

    if (PeriodTicks > 0) {
      // The register specifies the maximum value, so the period is really one tick longer
      ARR_RegisterValue = PeriodTicks - 1;
    } else {
      // But do not underflow in case a zero period was given somehow.
      ARR_RegisterValue = 0;
    }

Alternatively, you could maybe trigger an error when the period would end up as 0 ticks (which means that either 0 microsecond, 0 ticks, or a frequency higher than the input clock was specified). But maybe just using a register value of 0 ticks (i.e. run as fast as possible even if not as fast as requested, as your PR does) is the better approach.

Reply:
I prefer 1st option as error check, is not always performed at application level.
Done